### PR TITLE
chore: replace usage of `ParseDenomTrace`

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

The remaining places where it is used should be removed with https://github.com/cosmos/ibc-go/issues/6414

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of coin denominations to ensure accurate processing and validation.

### CODE REVIEW

**Comment by colin-axner** [comment]
heads up, I'm working on a pr to remove `ParseDenomTrace` so we might cover same code paths

**Review by colin-axner — APPROVED** [review]

File: e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -518,8 +518,8 @@ func (s *E2ETestSuite) InitiateChannelUpgrade(ctx context.Context, chain ibc.Cha
 }
 
 // GetIBCToken returns the denomination of the full token denom sent to the receiving channel
-func GetIBCToken(fullTokenDenom string, portID, channelID string) transfertypes.DenomTrace {
-	return transfertypes.ParseDenomTrace(fmt.Sprintf("%s/%s/%s", portID, channelID, fullTokenDenom))
+func GetIBCToken(fullTokenDenom string, portID, channelID string) transfertypes.Denom {
```
Comment (damiannolan) [inline e2e/testsuite/testsuite.go:521]: Just a thought, will this change impact compatability tests based on ibc-go version? 

I'm assuming it should be fine as its just used for creating the `ibc/{hash}` denom.

**Review by damiannolan — APPROVED** [review]
Nice 👌

File: e2e/testsuite/testsuite.go (hunk 1 above)
Comment (colin-axner) [reply e2e/testsuite/testsuite.go:521]: No functional change

File: e2e/testsuite/testsuite.go [hunk 2]
```diff
@@ -518,8 +518,8 @@ func (s *E2ETestSuite) InitiateChannelUpgrade(ctx context.Context, chain ibc.Cha
...
+	return transfertypes.ExtractDenomFromFullPath(fmt.Sprintf("%s/%s/%s", portID, channelID, fullTokenDenom))
```
Comment (colin-axner) [inline e2e/testsuite/testsuite.go:522]: parsing is unnecessary here, we can simply construct the denom

**Comment by colin-axner** [comment]
no diffs, as changes got included in the migrate denomtraces to denom pr

**Pull request closed without merging**